app/controllers/business_entities_controller.rb

```diff
@@ -0,0 +1,110 @@
+class BusinessEntitiesController < ApplicationController
+  def index
+    authorize :business_entity
+    list_jurisdictions
+  end
+
+  def new
+    redirect_to office_business_entities_path if valid_business_entity || !jurisdiction
+    @business_entity = BusinessEntity.new
+    authorize @business_entity
+  end
+
+  def create
+    @business_entity = business_entity_service.build(business_entity_params)
+    authorize @business_entity
+    if @business_entity.save
+      redirect_to office_business_entities_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    authorize business_entity
+  end
+
+  def update
+    bes = business_entity_service
+    new_be = bes.check_update(business_entity_params)
+    authorize new_be
+
+    if bes.persist_update!(new_be)
+      redirect_to office_business_entities_path
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  helper_method def office
+    @office ||= Office.find(params[:office_id])
+  end
+
+  helper_method def business_entity
+    @business_entity ||= BusinessEntity.find(params[:id])
+  end
+
+  def valid_business_entity
+    be = BusinessEntity.find_by(office_id: params[:office_id],
+                                jurisdiction_id: params[:jurisdiction_id],
+                                valid_to: nil)
+    flash[:alert] = t('error_messages.create_be_exists', jurisdiction: be.jurisdiction.name,
+                                                         office: be.office.name) if be
+    be
+  end
```

Comment:
Here, I'd use **business_entity** instead of **be**. Partly because this is 'hidden' like here:

```ruby
flash[:alert] = t('error_messages.create_be_exists', jurisdiction: be.jurisdiction.name,
                                                     office: be.office.name) if be
```

Just because this line is so long, I'd maybe write it as:

```ruby
flash[:alert] = t('error_messages.create_be_exists', 
                         jurisdiction: be.jurisdiction.name,
                         office: be.office.name) if business_entity
```
But looking at it now, it also seems to 'hide' the **if** condition.

This might seem better:

```ruby
if business_entity
  flash[:alert] = t('...
end
business_entity
```

What do you think?